node/impls.go

```diff
@@ -148,52 +153,61 @@ func (l agreementLedger) EnsureValidatedBlock(ve agreement.ValidatedBlock, c agr
 }
 
 // EnsureDigest implements agreement.LedgerWriter.EnsureDigest.
+// This method should not block.
 // TODO: Get rid of EnsureDigest -- instead the ledger should expose what blocks it's waiting on, and a separate service should fetch them and call EnsureBlock
 // should "retry until cert matches" logic live here or in the abstract fetcher?
 func (l agreementLedger) EnsureDigest(cert agreement.Certificate, quit chan struct{}, verifier *agreement.AsyncVoteVerifier) {
 	round := cert.Round
 	blockHash := bookkeeping.BlockHash(cert.Proposal.BlockDigest) // semantic digest (i.e., hash of the block header), not byte-for-byte digest
 	logging.Base().Debug("consensus was reached on a block we don't have yet: ", blockHash)
-	for {
-		// Ask the fetcher to get the block somehow
-		block, fetchedCert, err := l.FetchBlockByDigest(round, quit)
-		if err != nil {
-			select {
-			case <-quit:
-				logging.Base().Debugf("EnsureDigest was asked to quit before we could acquire the block")
+
+	go func() {
```

Comment:
nit: for that sake of readability, could you move this into its own function rather than making an anonymous function ?